box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiFile.java

```diff
@@ -541,4 +555,89 @@ public BoxApiFile(BoxSession session) {
     public BoxRequestsFile.FilePreviewed getFilePreviewedRequest(String fileId) {
         return new BoxRequestsFile.FilePreviewed(fileId, getPreviewFileUrl(), mSession);
     }
+
+    /**
+     * Gets a request that creates an upload session for uploading a new file or a new file version
+     *
+     * @param  file The file to be uploaded
+     * @param folderId Folder Id
+     * @return request to inform the server that a file was previewed
+     */
+    public BoxRequestsFile.CreateUploadSession getCreateUploadSessionRequest(File file, String folderId)
+            throws FileNotFoundException {
+        return new BoxRequestsFile.CreateUploadSession(file, folderId, getUploadSessionsUrl(), mSession);
+    }
+
+    /**
+     * Get a request for uploading a part to a BoxUploadSession
+     * @param file the file that is used to read the chunk using the ranges for the partnumber
+     * @param partNumber The part number being uploaded
+     * @param uploadSession the BoxUploadSession
+     * @return
+     */
+    public BoxRequestsFile.UploadSessionPart getUploadSessionPartRequest(File file, BoxUploadSession uploadSession,
+                                                                         int partNumber) throws IOException {
+        return new BoxRequestsFile.UploadSessionPart(file, uploadSession, partNumber, mSession);
+    }
+
+    /**
+     * Commit an upload session after all parts have been uploaded, creating the new file or the version
+     * @param uploadedParts the list of uploaded parts to be committed.
+     * @param attributes the key value pairs of attributes from the file instance.
+     * @param ifMatch ensures that your app only alters files/folders on Box if you have the current version.
+     * @param ifNoneMatch ensure that it retrieve unnecessary data if the most current version of file is on-hand.
+     * @return the created file instance.
+     */
+    public BoxRequestsFile.CommitUploadSession getCommitPartRequest(List<BoxUploadSessionPart> uploadedParts,
+                                                         Map<String, String> attributes,
+                                                         String ifMatch, String ifNoneMatch, BoxUploadSession uploadSession) {
+        return new BoxRequestsFile.CommitUploadSession(uploadedParts, attributes, ifMatch, ifNoneMatch, uploadSession, mSession);
+    }
+
+    /**
+     * Commit an upload session after all parts have been uploaded, creating the new file or the version
+     * @param uploadedParts the list of uploaded parts to be committed.
+     * @return the created file instance.
+     */
+    public BoxRequestsFile.CommitUploadSession getCommitPartRequest(List<BoxUploadSessionPart> uploadedParts, BoxUploadSession uploadSession) {
+        return new BoxRequestsFile.CommitUploadSession(uploadedParts, null, null, null, uploadSession, mSession);
+    }
+
+    /**
+     * Gets the URL for file information
```

Comment:
I believe this API is to get the URL for the session information (as it is used below), so a few things:
 - Please rename the Id param to be sessionId
 - Update JavaDoc as it does not reflect the real usage of this API